tests/Fixer/FunctionNotation/VoidReturnFixerTest.php

```diff
@@ -242,4 +243,52 @@ public function provideFixPhp74Cases()
             ],
         ];
     }
+
+    /**
+     * Test if magic method is handled without causing syntax error.
+     *
+     * @dataProvider provideMethodWillNotCauseSyntaxErrorCases
+     *
+     * @param string $method
+     * @param int    $arguments
+     * @param bool   $static
+     */
+    public function testMethodWillNotCauseSyntaxError($method, $arguments = 0, $static = false)
+    {
+        $tokens = Tokens::fromCode(sprintf(
+            '<?php class Test { public%s function %s(%s) {} }',
+            $static ? ' static' : '',
+            $method,
+            implode(',', array_map(
+                function ($n) { return sprintf('$x%d', $n); },
+                array_keys(array_fill(0, $arguments, true))
+            ))
+        ));
+
+        $this->fixer->fix($this->getTestFile(), $tokens);
+
+        static::assertNull($this->lintSource($tokens->generateCode()));
+    }
+
+    public static function provideMethodWillNotCauseSyntaxErrorCases()
+    {
+        // List: https://www.php.net/manual/en/language.oop5.magic.php
+        yield ['__construct'];
+        yield ['__destruct'];
+        yield ['__call', 2];
+        yield ['__callStatic', 2, true];
+        yield ['__get', 1];
+        yield ['__set', 2];
+        yield ['__isset', 1];
+        yield ['__unset', 1];
+        yield ['__sleep'];
+        yield ['__wakeup'];
+        yield ['__serialize'];
+        yield ['__unserialize', 1];
+        yield ['__toString'];
```

Comment:
should be fixed to `__toString(): string`

Reply:
no, it should not

Reply:
Not by this fixer, though.